src/resolve/source.rs

```diff
@@ -198,11 +257,29 @@ fn to_include(
     }
 }
 
-fn to_path<P: AsRef<Path>>(path: &str, workdir: P) -> io::Result<PathBuf> {
-    let path = Path::new(&path);
-    let old_workdir = env::current_dir()?;
-    env::set_current_dir(workdir)?;
-    let path = path.canonicalize()?;
-    env::set_current_dir(old_workdir)?;
-    Ok(path)
+fn to_path<P: AsRef<Path>>(url: &Url, workdir: P) -> std::result::Result<PathBuf, PathError> {
+    let mut full_path = workdir.as_ref().to_path_buf();
+
+    url
+        .path_segments()
+        .ok_or(PathError::NoPath)?
```

Comment:
I can specify an include `![](heradoc://document/)`, which will result in an `InternalError`. I don't think that should be the case.

Reply:
What should that resolve to? That sounds like a good test.

Reply:
Also, I had assumed the path to be `/` in that case. Seems like the assumption was wrong, then this may deserve its own variant.

Reply:
Actually both `heradoc://document/` and `heradoc://document` have a path segment `""`: <https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=597179843b1f2dce6824cd357c06b45f>. So I guess we're fine here. I guess if `NoPath` can even happen here? If not, we should probably just `expect` it instead of possibly handling an impossible case.

Reply:
Good. It's not intended to ever occur, so that's why it would result in an `bug` message then. I'll add the `url` in question to the `Error` variant so that we know how this occured when a bug report is opened.